WordPress/Classes/Services/SiteAssemblyService.swift

```diff
@@ -122,25 +118,31 @@ final class EnhancedSiteCreationService: LocalCoreDataService, SiteAssemblyServi
 
                     return
                 }
-
-                self.synchronize(createdSite: response.createdSite)
+                self.coreDataStack.performAndSave({ context in
+                    self.createSite(for: response.createdSite, in: context)
+                }, completion: { [weak self] blogID in
+                    guard let blogID else {
+                        self?.endFailedAssembly()
+                        return
+                    }
+                    self?.syncBlogAndAccount(createdBlogID: blogID)
+                }, on: .main)
             case .failure(let creationError):
                 DDLogError("\(creationError)")
                 self.endFailedAssembly()
             }
         }
     }
 
-    private func synchronize(createdSite: CreatedSite) {
-        guard let defaultAccount = try? WPAccount.lookupDefaultWordPressComAccount(in: managedObjectContext) else {
-            endFailedAssembly()
-            return
+    private func createSite(for createdSite: CreatedSite, in context: NSManagedObjectContext) -> NSManagedObjectID? {
+        guard let defaultAccount = try? WPAccount.lookupDefaultWordPressComAccount(in: context) else {
```

Comment:
Would it help error handling (or let's say logging) if we re-threw the error here instead of using `.none` as a fallback?

Reply:
It appears there isn't much error handling in this type here—all error cases end up calling `endFailedAssembly()`.